.github/workflows/integration_tests.yml

```diff
@@ -115,8 +97,10 @@ jobs:
         set -e
         echo Running test.sh
         ./src/ci/bin/test.sh
+    # always() is some github magic that forces the following step to run, even when the previous fails.
+    # Without it, the if statement won't be evaluated on a test failure.
     - uses: ravsamhq/notify-slack-action@v2
-      if: inputs.slack-alert
+      if: always() && github.ref == 'refs/heads/develop' #only report on failures against develop.
```

Comment:
Fascinating. Thank you very much for including this comment, it documents an essential but non-obvious aspect of the test.